### PR TITLE
Add fetch to get all tickets

### LINKED ISSUE #321: Create fetch to GET list Tickets
## Add fetch to get all tickets

We have added the functionality to load all tickets from the system so they can be displayed in the application.  
The feature ensures that users can see their tickets and any updates in a reliable way.  
We also made sure that possible errors or connection issues are handled properly to avoid breaking the user experience.

### PR DESCRIPTION
**Pull request by albertvallsbe**
## Add fetch to get all tickets

Implemented a custom React hook to fetch all tickets using Axios with proper authentication headers.  
Handled API response validation, including success flags and invalid response shapes.  
Added comprehensive error handling for Axios errors, network failures, and request cancellations.  
Included full test coverage for success, edge cases, and failure scenarios to ensure robustness.

### CODE REVIEW

**Review by vrsirvent — APPROVED** [review]

**Review by triflip — APPROVED** [review]
Good job! The hook is well structured, with good error handling, response shape validation, and correct use of AbortController.
The tests cover all important cases, and the types are very well defined.
No blocking comments from my side.

**Review by kenan-it-academy — CHANGES_REQUESTED** [review]
Will review again later but 362 lines added for a *single endpoint* seems wild to me 😱 
Something is deeply wrong here.
At the very least there is a heap of duplicated code with the rest of the endpoint hooks, this needs to be refactored ASAP. Add an urgent backlog task for that, please.

File: frontend/src/hooks/__tests__/useTicketingGetAll.test.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+    await waitFor(() => {
+      expect(result.current.isLoading).toBe(false);
+    });
+
+    expect(result.current.tickets).toEqual([]);
+    expect(result.current.errorMessage).toBe("Unauthorized");
+  });
+
+  it("sets errorMessage when API response shape is invalid", async () => {
```
Comment (kenan-it-academy) [inline frontend/src/hooks/__tests__/useTicketingGetAll.test.ts]: Out of scope

File: frontend/src/hooks/__tests__/useTicketingGetAll.test.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+    await waitFor(() => {
+      expect(result.current.isLoading).toBe(false);
+    });
+
+    expect(result.current.tickets).toEqual([]);
+    expect(result.current.errorMessage).toBe("Invalid API response shape");
+  });
+
+  it("sets errorMessage on AxiosError with API message", async () => {
```
Comment (kenan-it-academy) [inline frontend/src/hooks/__tests__/useTicketingGetAll.test.ts]: out of scope

File: frontend/src/hooks/__tests__/useTicketingGetAll.test.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    await waitFor(() => {
+      expect(result.current.isLoading).toBe(false);
+    });
+
+    expect(result.current.tickets).toEqual([]);
+    expect(result.current.errorMessage).toBe("Network down");
+  });
+
+  it("ignores cancelled request and does not set errorMessage", async () => {
```
Comment (kenan-it-academy) [inline frontend/src/hooks/__tests__/useTicketingGetAll.test.ts]: out of scope

File: frontend/src/hooks/__tests__/useTicketingGetAll.test.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+  beforeEach(() => {
+    vi.clearAllMocks();
+    vi.spyOn(Storage.prototype, "getItem").mockReturnValue("fake-token");
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  it("returns tickets when API success=true", async () => {
```
Comment (kenan-it-academy) [inline frontend/src/hooks/__tests__/useTicketingGetAll.test.ts]: follow DRY principles and extract duplicated test code into a function

File: frontend/src/hooks/useTicketingGetAll.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+              Accept: "application/json",
+              Authorization: `Bearer ${token}`,
+            },
+            signal: abortController.signal,
+          },
+        );
+
+        const responseData = response.data;
+
+        if (
```
Comment (kenan-it-academy) [inline frontend/src/hooks/useTicketingGetAll.ts]: readabilty: hard to understand what case this is.

Also: out of scope 😇

File: frontend/src/hooks/useTicketingGetAll.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+          !responseData ||
+          typeof responseData.success !== "boolean" ||
+          !Array.isArray(responseData.data)
+        ) {
+          throw {
+            message: "Invalid API response shape",
+          } as TicketingError;
+        }
+
+        if (!responseData.success) {
```
Comment (kenan-it-academy) [inline frontend/src/hooks/useTicketingGetAll.ts:44]: Out of... you get the idea

File: frontend/src/hooks/useTicketingGetAll.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
+import axios, { AxiosError, CanceledError } from "axios";
```
Comment (kenan-it-academy) [inline frontend/src/hooks/useTicketingGetAll.ts:1]: Add to backlog: create a `UseEndpoint` hook that exctracts all duplication from the Endpoint hooks because OH MY GOD I WANT TO DIE

**Review by kenan-it-academy — CHANGES_REQUESTED** [review]

**Review by kenan-it-academy — APPROVED** [review]
Still big, but kind of the useHook templat's fault. Add to backlog that we need to extract a useEndpoint hook so next person doesn't want to commit sudoku.

**Comment by albertvallsbe** [comment]
@kenan-it-academy The task in backlog for a useEndpoind hook was added today!
